core/src/main/java/ai/timefold/solver/core/impl/score/director/InnerScore.java

```diff
@@ -0,0 +1,56 @@
+package ai.timefold.solver.core.impl.score.director;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.api.domain.variable.PlanningListVariable;
+import ai.timefold.solver.core.api.domain.variable.PlanningVariable;
+import ai.timefold.solver.core.api.score.Score;
+
+/**
+ * Carries information on if the {@link PlanningSolution} of this score was fully initialized when it was calculated.
+ * This only works for solutions where:
+ * <ul>
+ * <li>{@link PlanningVariable basic variables} are used,
+ * and {@link PlanningVariable#allowsUnassigned() unassigning} is not allowed.</li>
+ * <li>{@link PlanningListVariable list variables} are used,
+ * and {@link PlanningListVariable#allowsUnassignedValues() unassigned values} are not allowed.</li>
+ * </ul>
+ *
+ * For solutions which do allow unassigning values, {@link #unassignedCount} is always zero.
+ */
+public record InnerScore<Score_ extends Score<Score_>>(Score_ raw, int unassignedCount)
+        implements
+            Comparable<InnerScore<Score_>> {
+
+    public static <Score_ extends Score<Score_>> InnerScore<Score_> of(Score_ score) {
+        return ofUninitialized(score, 0);
+    }
+
+    public static <Score_ extends Score<Score_>> InnerScore<Score_> ofUninitialized(Score_ score, int uninitializedCount) {
```

Comment:
Can this method be called for a model that accepts unassigned values? If that is the case, it should not be considered as an uninitialized solution.